Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.h

```diff
@@ -63,6 +63,42 @@ extern NSString * const MTLBooleanValueTransformerName;
 /// with a default value of `nil` and a reverse default value of `nil`.
 + (NSValueTransformer<MTLTransformerErrorHandling> *)mtl_valueMappingTransformerWithDictionary:(NSDictionary *)dictionary;
 
+/// A reversible value transformer to transform between a date and its string
+/// representation
+///
+/// dateFormat - The date format used by the date formatter (http://www.unicode.org/reports/tr35/tr35-31/tr35-dates.html#Date_Field_Symbol_Table)
+/// calendar   - The calendar used by the date formatter
+/// locale     - The locale used by the date formatter
+/// timeZone   - The time zone used by the date formatter
+///
+/// Returns a transformer which will map from strings to dates for forward
+/// transformations, and from dates to strings for reverse transformations.
++ (NSValueTransformer<MTLTransformerErrorHandling> *)mtl_dateTransformerWithDateFormat:(NSString *)dateFormat calendar:(NSCalendar *)calendar locale:(NSLocale *)locale timeZone:(NSTimeZone *)timeZone defaultDate:(NSDate *)defaultDate;
+
+/// Returns a value transformer created by calling
+/// `+mtl_dateTransformerWithDateFormat:calendar:locale:timeZone:defaultDate:`
+/// with a calendar, locale, time zone and default date of `nil`.
++ (NSValueTransformer<MTLTransformerErrorHandling> *)mtl_dateTransformerWithDateFormat:(NSString *)dateFormat;
+
+/// A reversible value transformer to transform between a number and its string
+/// representation
+///
+/// numberStyle - The number style used by the number formatter
+///
+/// Returns a transformer which will map from strings to numbers for forward
+/// transformations, and from numbers to strings for reverse transformations.
++ (NSValueTransformer<MTLTransformerErrorHandling> *)mtl_numberTransformerWithNumberStyle:(NSNumberFormatterStyle)numberStyle;
```

Comment:
Should we force the user to pass in a locale? Otherwise, things like decimal separator might be different for different users (I don't know from the top of my head, but I imagine it to default to the current locale)